geode-core/src/test/java/org/apache/geode/internal/cache/partitioned/PersistentColocatedPartitionedRegionDUnitTest.java

```diff
@@ -1228,6 +1226,7 @@ public Object call() throws Exception {
           try {
             // Another delay before exiting the thread to make sure that missing region logging
             // doesn't continue after all regions are created (delay > logInterval)
+            verify(mockAppender, atLeastOnce()).append(any(LogEvent.class));
```

Comment:
With this it looks like we are going to get 2 log events...I am wondering if its due to long poll interval time with the first log event check...That is adding up more time, which causes second log message to be generated....Will it help if we change the first log wait time to have a small poll interval: with().pollInterval(POLL_INTERVAL_MS, TimeUnit.MILLISECONDS).await()
          .atMost(TEST_WAIT_MS, TimeUnit.MILLISECONDS).until(()

Reply:
The default poll interval in 100ms. It is possible that decreasing that could narrow or close the window for the bug, but still does not inherently prevent it and so the race condition would still be present, and would still depend on how long createPR(...) takes. My solution would get a second log event in the case where the existing race condition would be present, but still tests the desired functionality correctly: that is, that the creation of the partitioned region silences the warning. If the warning message is still being generated after createPR(...) completes, then the test will fail.